### PR TITLE
Face: Feature - Layered image

### PR DESCRIPTION
**Pull request by MalcolmnDEV**

### CODE REVIEW

File: BlockV/Face/Face Views/Layered Image/LayeredImageFaceView.swift [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+		// ensure base has correct bounds with the 'parent' vAtom
+		baseLayer.frame = self.bounds
+		baseLayer.vatom = vatom
+        self.addSubview(baseLayer)
+
+		// initial setup
+		self.vAtomStateChanged()
+
+		// listen to websocket for state changes on LayeredImageView
```
Comment (cjmconie) [inline BlockV/Face/Face Views/Layered Image/LayeredImageFaceView.swift]: Please remove comments.

File: BlockV/Face/Face Views/FaceViewRoster.swift [hunk 2]
```diff
@@ -48,7 +48,10 @@ public class FaceViewRoster {
         let roster = FaceViewRoster()
         // embedded face views
         roster.register(ImageFaceView.self)
+        roster.register(ImagePolicyFaceView.self)
         roster.register(ProgressImageFaceView.self)
+		roster.register(LayeredImageFaceView.self)
```
Comment (cjmconie) [inline BlockV/Face/Face Views/FaceViewRoster.swift]: There is some odd spacing still - swiftlint is your friend :)

File: BlockV/Face/Face Views/Layered Image/LayeredImageFaceView.swift [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+		}
+	}
+
+	/// Fetch the children for the specified vAtom identifier and apply the relevant layers.
+
+	private func vAtomStateChanged() {
+		BLOCKv.getInventory(id: self.vatom.id) { (vatomModels, error) in
+
+			guard error == nil else {
+				printBV(info: "getInventory - \(error!.localizedDescription)")
```
Comment (cjmconie) [inline BlockV/Face/Face Views/Layered Image/LayeredImageFaceView.swift]: Please use `printBV(error:)` for errors

File: BlockV/Face/Face Views/Layered Image/LayeredImageFaceView.swift [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+import Foundation
+import Nuke
+
+/// Native Layered face view
+class LayeredImageFaceView: FaceView {
+    class var displayURL: String { return "native://layered-image" }
+
+	// Layer must be a class to inherit from UIImageview
+	class Layer: UIImageView {
+		// Reference to the resource this layer displays.
```
Comment (cjmconie) [inline BlockV/Face/Face Views/Layered Image/LayeredImageFaceView.swift]: Both `VatomResourceModel` and `VatomModel` are structs - so these are not strictly "references" . Prefer:

// Resource this layer displays
// vAtom this layer displays.

File: BlockV/Face/Face Views/Layered Image/LayeredImageFaceView.swift [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) should not be called on Face Views. Please use VatomView.")
+    }
+
+	// MARK: - View Lifecylce
+
+	override func layoutSubviews() {
+		super.layoutSubviews()
+		updateContentMode(forLayer: baseLayer)
```
Comment (cjmconie) [inline BlockV/Face/Face Views/Layered Image/LayeredImageFaceView.swift]: Why does the content mode for the layer need to be updated on `layoutSubviews`? This property need only be set once, right?

Comment (MalcolmnDEV) [reply BlockV/Face/Face Views/Layered Image/LayeredImageFaceView.swift]: Yes, it was there when config still have fill and fit

**Pull request merged**